mirascope/core/gemini/_utils/_setup_call.py

```diff
@@ -43,18 +43,22 @@ def setup_call(
     messages = cast(list[BaseMessageParam | ContentDict], messages)
     messages = convert_message_params(messages)
     if json_mode:
-        generation_config = call_kwargs.get("generation_config", {})
-        generation_config["response_mime_type"] = "application/json"
-        call_kwargs["generation_config"] = generation_config
```

Comment:
wait so was this just not working as all previously? I'm confused cause I feel like this was working

Reply:
we should update the `GeminiCallParams` type to support `GenerationConfigType`, and then here if it turns out to be a dataclass we should handle that (i.e. convert to dict) and then the rest of this existing code should work as intended

Reply:
Fixed